api-docs/paths/dataSet.yaml

```diff
@@ -0,0 +1,62 @@
+parameters:
+  - $ref: "../api.yaml#/parameters/dataSetId"
+
+get:
+  summary: Get metadata
+  description: >
+        Get the name, description, record count, and other information
+        about the given data set. Does not return any records within the
+        data set.
+  operationId: findDataSetById
+  tags: ["DataSet"]
```

Comment:
I think we should use the same "DataSets" tag here so the operations are grouped together in the documentation

Reply:
These are technically two different paths - one for /api/datasets and the other for /api/datasets/{datasetId} .. there may be a method that conflicts as a result if we were to group them .. for instance just the GET .. but perhaps that's ok if they have different summaries?